use_case_examples/hybrid_model/serve_model.py

```diff
@@ -0,0 +1,193 @@
+"""Hybrid Model Deployment Server.
+
+Routes:
+    - Get all names
+    - Get client.zip
+    - Add a key
+    - Compute
+"""
+
+import ast
+import io
+import os
+import time
+import uuid
+from collections import defaultdict
+from functools import lru_cache
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+import uvicorn
+from fastapi import FastAPI, Form, HTTPException, UploadFile
+from fastapi.responses import FileResponse, StreamingResponse
+from loguru import logger
+
+# No relative import here because when not used in the package itself
+from concrete.ml.deployment import FHEModelServer
+
+
+def underscore_str_to_tuple(tup):
+    return ast.literal_eval(tup.replace("p_", "(").replace("_p", ")").replace("_", ", "))
+
+
+if __name__ == "__main__":
+    app = FastAPI(debug=False)
+    # Model-name -> Module-Name -> Input-shape
+
+    FILE_FOLDER = Path(__file__).parent
+    KEY_PATH = Path(os.environ.get("KEY_PATH", FILE_FOLDER / Path("user_keys")))
+    KEY_PATH.mkdir(exist_ok=True)
+    MODELS_PATH = Path(os.environ.get("PATH_TO_MODELS", FILE_FOLDER / Path("model")))
+    PORT = os.environ.get("PORT", "5000")
+    MODULES = defaultdict(dict)
+    # Populate modules -> could be done dynamically on each query tbh
+    for model_path in MODELS_PATH.iterdir():  # Model
+        # TODO: change with a struct/obj
+        model_name = model_path.name
+        MODULES[model_name] = defaultdict(dict)
+        for module_path in model_path.iterdir():  # Module
+            if not module_path.is_dir():
+                continue
+            module_name = module_path.name
+            MODULES[model_name][module_name] = defaultdict(dict)
+            for input_shape_path in module_path.iterdir():
+                if not input_shape_path.is_dir():
+                    continue
+                input_shape = str(underscore_str_to_tuple(input_shape_path.name))
+                MODULES[model_name][module_name][input_shape] = {
+                    "path": input_shape_path.resolve(),
+                    "module_name": module_name,
+                    "model_name": model_name,
+                    "shape": input_shape,
+                }
+
+    @lru_cache(maxsize=None)
```

Comment:
what is this ?

Reply:
This is a way to cache the result of a function.
Avoids re-computing the function.

Reply:
ok didn't know about it ! is it safe to cache keys like this ? I mean here it's just a use case example but maybe not a good practice to show ?

Reply:
What would be unsafe about this?

Reply:
well not exactly sure I was just thinking that CP's caching key features is considered an unsafe feature , so there might be some reasons

Reply:
these are boostrapping keys right ?

Reply:
These are eval keys.

Unsafe because they are cached in the file-system where things might change, here we are talking in memory caching.

Reply:
ok sure then, fine for me !